command/setup_consul.go

```diff
@@ -0,0 +1,485 @@
+
+	flags := s.Meta.FlagSet(s.Name(), FlagSetClient)
+	flags.Usage = func() { s.Ui.Output(s.Help()) }
+	flags.BoolVar(&s.autoYes, "y", false, "")
+	flags.StringVar(&s.jwksURL, "jwks-url", "http://localhost:4646/.well-known/jwks.json", "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments.
+	if len(flags.Args()) != 0 {
+		s.Ui.Error("This command takes no arguments")
+		s.Ui.Error(commandErrorText(s))
+		return 1
+	}
+
+	var err error
+
+	s.Ui.Output(`
+This command will walk you through configuring all the components required for 
+Nomad workloads to authenticate themselves against Consul ACL using their 
+respective workload identities. 
+
+First we need to connect to Consul. 
+`)
+
+	cfg := api.DefaultConfig()
+	if !s.autoYes {
+		if !s.askQuestion(fmt.Sprintf("Is %q the correct address of your Consul cluster? [Y/n]", cfg.Address)) {
+			s.Ui.Warn(`
+Please set the CONSUL_HTTP_ADDR variable to your Consul cluster address and re-run the command.`)
+			return 0
+		}
+	}
+
+	// Get the Consul client.
+	s.client, err = api.NewClient(cfg)
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing Consul client: %s", err))
+		return 1
+	}
+
+	// check if we're connecting to Consul ent
+	if _, err := s.client.Operator().LicenseGet(nil); err == nil {
+		s.consulEnt = true
+	}
+
+	authMethodMsg := `
+We need to create two JWT auth methods: one for Nomad services, and one for
+Nomad tasks. The method for services will be called %q and
+the method for tasks %q, and they will both be of jwt type.
+
+They will share the following config:`
+	s.Ui.Output(fmt.Sprintf(authMethodMsg, authMethodServices, authMethodTasks))
+
+	authMethodConf, err := s.renderAuthMethodConf()
+	if err != nil {
+		s.Ui.Error(err.Error())
+		return 1
+	}
+
+	jsConf, _ := json.MarshalIndent(authMethodConf, "", "    ")
+	s.Ui.Output(string(jsConf))
+
+	if s.consulEnt {
+		namespaceMsg := `
+Since you're running Consul Enterprise, we will additionally create
+a namespace %s and bind the auth methods to that namespace.
+`
+		s.Ui.Output(fmt.Sprintf(namespaceMsg, consulNamespace))
+	}
+
+	var createAuthMethods bool
+	if !s.autoYes {
+		createAuthMethods = s.askQuestion("Create these auth methods in your Consul cluster? [Y/n]")
+	} else {
+		createAuthMethods = true
+	}
+
+	if s.consulEnt {
+		err = s.createNamespace()
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	if createAuthMethods {
+		err = s.createAuthMethod(authMethodServices, authMethodConf)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+		err = s.createAuthMethod(authMethodTasks, authMethodConf)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	servicesBindingRule := &api.ACLBindingRule{
+		Description: "binding rule for Nomad services w/ (WI)",
+		AuthMethod:  authMethodServices,
+		BindType:    "service",
+		BindName:    "${value.nomad_namespace}-${value.nomad_service}",
```

Comment:
```suggestion
		BindName:    "${value.nomad_service}",
```

Explained in https://github.com/hashicorp/nomad/pull/18820#discussion_r1373773861.